dm/worker/config_test.go

```diff
@@ -14,20 +14,49 @@
 package worker
 
 import (
+	"path"
+	"strings"
+
 	. "github.com/pingcap/check"
 )
 
-func (t *testWorker) TestConfig(c *C) {
-	cfg := &Config{}
+func (t *testServer) TestConfig(c *C) {
+	cfg := NewConfig()
 
-	err := cfg.configFromFile("./dm-worker.toml")
-	c.Assert(err, IsNil)
+	c.Assert(cfg.Parse([]string{"-config=./dm-worker.toml"}), IsNil)
 	c.Assert(cfg.SourceID, Equals, "mysql-replica-01")
 
+	dir := c.MkDir()
+	cfg.ConfigFile = path.Join(dir, "dm-worker.toml")
+
+	// test clone
 	clone1 := cfg.Clone()
 	c.Assert(cfg, DeepEquals, clone1)
+	clone1.SourceID = "xxx"
+	c.Assert(cfg.SourceID, Equals, "mysql-replica-01")
+
+	// test format
+	c.Assert(strings.Contains(cfg.String(), "mysql-replica-01"), IsTrue)
+	tomlStr, err := clone1.Toml()
+	c.Assert(err, IsNil)
+	c.Assert(strings.Contains(tomlStr, "xxx"), IsTrue)
+	originCfgStr, err := cfg.Toml()
+	c.Assert(err, IsNil)
+	c.Assert(strings.Contains(tomlStr, "mysql-replica-01"), IsTrue)
```

Comment:
`tomlStr` seems not updated after L40, why test this here?

Reply:
it's a mistake